### ISSUE TITLE
GitHub Actions: ':app:lintVitalRelease'. & then  ':app:lintVitalRelease'.

### ISSUE THREAD

**Issue by kenaitian** [flutter/flutter#105393]
1. The Flutter app loads and runs locally on Android Studio (Flutter) but not on GitHub Actions per the log below:
[https://github.com/lessimp/lessimp_logs.git](https://www.google.com/url?q=https://github.com/lessimp/lessimp_logs.git&sa=D&source=buganizer&usg=AOvVaw3vIS1W6qtIqXJlcZB4x7mr)

2. The app continues to load and run locally when I suppress the errors under "buildTypes { release {..." of android/app/build.grade and any where else there's lintOptions as follows:
```gradle
    buildTypes {
        release {
            lintOptions {
                disable 'InvalidPackage'
                disable "Instantiatable"
                checkReleaseBuilds false
                abortOnError false
            }
            signingConfig signingConfigs.release
        }
    }
```
4. GitHub Action results after Step 2 even though "signingConfig signingConfigs.release" is set above as well as that encrypted Keystore file (gpg) that gets decrypted via GitHub Action with GitHub Secret: "Execution failed for task ':app:validateSigningRelease'.

Run flutter build apk --release
 ```
💪 Building with sound null safety 💪
Running Gradle task 'assembleRelease'...                        
/opt/hostedtoolcache/flutter/3.0.1-stable/x64/.pub-cache/hosted/[pub.dartlang.org/mobile_scanner-1.1.2/lib/src/mobile_scanner.dart:50:20](https://www.google.com/url?q=http://pub.dartlang.org/mobile_scanner-1.1.2/lib/src/mobile_scanner.dart:50:20&sa=D&source=buganizer&usg=AOvVaw2LXJrWuACV55-Bietfyf8b): Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/opt/hostedtoolcache/flutter/3.0.1-stable/x64/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance?.addObserver(this);
                   ^
/opt/hostedtoolcache/flutter/3.0.1-stable/x64/.pub-cache/hosted/[pub.dartlang.org/mobile_scanner-1.1.2/lib/src/mobile_scanner.dart:134:20](https://www.google.com/url?q=http://pub.dartlang.org/mobile_scanner-1.1.2/lib/src/mobile_scanner.dart:134:20&sa=D&source=buganizer&usg=AOvVaw0VhbrkT5GMxdxiAq9QNRGq): Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/opt/hostedtoolcache/flutter/3.0.1-stable/x64/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance?.removeObserver(this);
                   ^
Note: /opt/hostedtoolcache/flutter/3.0.1-stable/x64/.pub-cache/hosted/[pub.dartlang.org/firebase_crashlytics-2.8.1/android/src/main/java/io/flutter/plugins/firebase/crashlytics/FlutterFirebaseCrashlyticsPlugin.java](https://www.google.com/url?q=http://pub.dartlang.org/firebase_crashlytics-2.8.1/android/src/main/java/io/flutter/plugins/firebase/crashlytics/FlutterFirebaseCrashlyticsPlugin.java&sa=D&source=buganizer&usg=AOvVaw1S6zBW5hUkf5HYksX6I7PR) uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /opt/hostedtoolcache/flutter/3.0.1-stable/x64/.pub-cache/hosted/[pub.dartlang.org/location-4.4.0/android/src/main/java/com/lyokone/location/FlutterLocation.java](https://www.google.com/url?q=http://pub.dartlang.org/location-4.4.0/android/src/main/java/com/lyokone/location/FlutterLocation.java&sa=D&source=buganizer&usg=AOvVaw3fhJ0s1WFoN8LwFPiEiRv9) uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:validateSigningRelease'.
> Keystore file not set for signing config release
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at [https://help.gradle.org](https://www.google.com/url?q=https://help.gradle.org&sa=D&source=buganizer&usg=AOvVaw3j5UpJ9ZNA-E_PeJBUtgvu)
BUILD FAILED in 4m 43s
Running Gradle task 'assembleRelease'...                          283.9s
Gradle task assembleRelease failed with exit code 1
Error: Process completed with exit code 1.
```
These efforts seem to want to lead somewhere but so far to no end.

If necessary, I can start a collaboration session with any capable person on here through either an Invite or Collaborate via VS Live Share: Real-time Collaborative Development.

Click here for the [Collaboration session URL](https://prod.liveshare.vsengsaas.visualstudio.com/join?C0A95E33EB7BA3D4CD5A98C85EFB8EEF4AAD):

**Comment by kenaitian** [comment]
Solution/Error:
1. commented out the only lintOptions under "buildTypes { release {..." of android/app/build.grade as follows:
    buildTypes {
        release {
#            lintOptions {
#                disable 'InvalidPackage'
#                disable "Instantiatable"
#                checkReleaseBuilds false
#                abortOnError false
#            }
            signingConfig signingConfigs.release
        }
    }

2. Made sure my keys.properties and key.jks files were inside the android/app folder since the ':app:lintVitalRelease' is related to some path via /home/runner/work/lessimp/lessimp/android/app
keys.properties syntax where all others failed:
     storePassword=XXXXXX
     keyPassword=XXXXXX
     keyAlias=gpgKey
     storeFile=key.jks

3. Two jobs were expected from the android-cd.yml file:
a. Building Android --> SUCCEEDED
b. Deploying to playstore --> FAILED: 
<img width="1279" alt="Screen Shot 2022-06-05 at 6 32 20 PM" src="https://user-images.githubusercontent.com/63857624/172073740-0259629b-9df4-4ba4-9ce0-3f7fe4b9fbbe.png">

Run Fastlane:
Run maierj/fastlane-action@v1.4.0
  with:
    lane: closed_beta
    subdirectory: android
  env:
    JAVA_HOME_12.0.2_x64: /opt/hostedtoolcache/jdk/12.0.2/x64
    JAVA_HOME: /opt/hostedtoolcache/jdk/12.0.2/x64
    JAVA_HOME_12_0_2_X64: /opt/hostedtoolcache/jdk/12.0.2/x64
    FLUTTER_ROOT: /opt/hostedtoolcache/flutter/3.0.1-stable/x64
    PUB_CACHE: /opt/hostedtoolcache/flutter/3.0.1-stable/x64/.pub-cache
Executing lane closed_beta on Linux.
<html><head>
<meta http-equiv="content-type" content="text/html;charset=utf-8">
<title>500 Server Error</title>
</head>
<body text=#000000 bgcolor=#ffffff>
<h1>Error: Server Error</h1>
<h2>The server encountered an error and could not complete your request.<p>Please try again in 30 seconds.</h2>
<h2></h2>
</body></html>
Moving to subdirectory android
Successfully installed bundler-2.3.15
Parsing documentation for bundler-2.3.15
Installing ri documentation for bundler-2.3.15
Done installing documentation for bundler after 0 seconds
1 gem installed
/usr/lib/ruby/2.7.0/rubygems.rb:275:in `find_spec_for_exe': Could not find 'bundler' (1.17.2) required by your /home/runner/work/lessimp/lessimp/android/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:1.17.2`
        from /usr/lib/ruby/2.7.0/rubygems.rb:294:in `activate_bin_path'
        from /usr/local/bin/bundle:23:in `<main>'
/usr/lib/ruby/2.7.0/rubygems.rb:275:in `find_spec_for_exe': Could not find 'bundler' (1.17.2) required by your /home/runner/work/lessimp/lessimp/android/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:1.17.2`
        from /usr/lib/ruby/2.7.0/rubygems.rb:294:in `activate_bin_path'
        from /usr/local/bin/bundle:23:in `<main>'
Error: Error: Executing lane closed_beta failed.
Error: Executing lane closed_beta failed.

4. YouTube Solution after Step 3b but it defeats the purpose for Fastlane with GitHub Actions: [https://youtu.be/VHzWQJKvol0](https://www.google.com/url?q=https://youtu.be/VHzWQJKvol0&sa=D&source=buganizer&usg=AOvVaw1g35ORUxJXccM_AoJwtVpC)

**Comment by maheshmnj** [comment]
Hi @kenaitian, Thanks for filing the issue. Please ensure that you have provided a proper path to keystore file in `key.properties` Does `flutter build apk --release` run fine locally?

**Comment by kenaitian** [comment]
@maheshmnj I already done that in step 2 above and after uncommenting lintOptions in step 1, issue seemed to have been fixed leading to new fastlane error per this  [github_action_log](https://github.com/lessimp/lessimp_logs.git)

**Comment by maheshmnj** [comment]
@kenaitian The error is in your GitHub action log is different from the original issue you reported, As per the Github action logs, The error seems related to fastlane due to missing file specified, I would suggest to ensure that the file is present in correct path and accessible to fastlane. 

>[17:59:56]: Could not find service account json file at path '/home/runner/work/lessimp/lessimp/android/service_account_key.json'
[17:59:56]: You passed invalid parameters to 'upload_to_play_store'.
[17:59:56]: Check out the error below and available options by running `fastlane action upload_to_play_store`

If everything is in place and if you are still experiencing the issue, I would suggest filing this issue to the fastlane repository. Closing as this is not an issue with flutter, In case you disagree feel free to write in the comments and we will reopen it.
